internal/pyxis/pyxis.go

```diff
@@ -0,0 +1,80 @@
+package pyxis
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/shurcooL/graphql"
+)
+
+const (
+	DefaultPyxisHost = "catalog.redhat.com/api/containers"
+)
+
+type HTTPClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+type PyxisClient struct {
+	Client    HTTPClient
+	PyxisHost string
+}
+
+func (p *PyxisClient) getPyxisGraphqlURL() string {
+	return fmt.Sprintf("https://%s/graphql/", p.PyxisHost)
+}
+
+func NewPyxisClient(pyxisHost string, httpClient HTTPClient) *PyxisClient {
+	return &PyxisClient{
+		Client:    httpClient,
+		PyxisHost: pyxisHost,
+	}
+}
+
+func (p *PyxisClient) FindOperatorIndices(ctx context.Context, organization string) ([]OperatorIndex, error) {
+	// our graphQL query
+	var query struct {
+		FindOperatorIndices struct {
+			OperatorIndex []struct {
+				OCPVersion   graphql.String `graphql:"ocp_version"`
+				Organization graphql.String `graphql:"organization"`
+				EndOfLife    graphql.String `graphql:"end_of_life"`
+			} `graphql:"data"`
+			Errors struct {
+				Status graphql.Int    `graphql:"status"`
+				Detail graphql.String `graphql:"detail"`
+			} `graphql:"error"`
+			Total graphql.Int
+			Page  graphql.Int
+			// filter to make sure we get exact results, end_of_life is a string, querying for `null` yields active OCP versions.
+		} `graphql:"find_operator_indices(filter:{and:[{organization:{eq:$organization}},{end_of_life:{eq:null}}]})"`
+	}
+
+	// variables to feed to our graphql filter
+	variables := map[string]interface{}{
+		"organization": graphql.String(organization),
+	}
+
+	// make our query
+	httpClient, ok := p.Client.(*http.Client)
+	if !ok {
+		return nil, fmt.Errorf("client could not be used as http.Client")
+	}
+	client := graphql.NewClient(p.getPyxisGraphqlURL(), httpClient)
```

Comment:
For clarity, in another comment I reference a type assertion - and this is the offending line.